### ISSUE TITLE
Refactor input-date-picker connectFocusTrap config options 

### ISSUE THREAD

**Issue by Elijbet** [Esri/calcite-design-system#10899]
### Description

Refactor `connectFocusTrap` config options to use callbacks to return focus based on whether it was closed via keyboard vs mouse. This change was made to avoid clicking on a separate, external element and having focus jump back to the `input-date-picker`.
 
https://github.com/Esri/calcite-design-system/commit/0d9ddc9018ff8c5eacee7592998189e59d512bf6#diff-4d4cdb442e767292afec664b7753872b8065094a8ddb8c19226d68126e156fa6R906
 
In the keyboard case, it will always fire on the element vs clicks.
 
After this refactor when `focusTrap` is enabled, it will take care of focus to return to text input when date picker is closed with escape. Refactor escape `keyDownHandler` to only close with escape when `focusTrapDisabled` is true.
 
https://github.com/Esri/calcite-design-system/pull/10578/files#diff-42c23e5a429472ae2ecb636c2ba383162681d40c5ff70c3f89be55d376052e32R799

### Proposed Advantages

Use of `focusTrap` util fully as intended and not have redundant component logic. 

### Which Component

input-date-picker

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

**Comment by Elijbet** [comment]
> [@Elijbet](https://github.com/Elijbet) Is this still valid once [#11625](https://github.com/Esri/calcite-design-system/pull/11625) lands?

We're still setting `setReturnFocus` to false in the new controller implementation here, which we did originally to ensure that clicking on external element will not have focus jump back to the input-date-picker, but prevents focus returning to text input when date picker is closed with escape. I think we're trying to make sure that enabled focusTrap does what's intended and returns the focus. And we only return the focus manually when focusTrapDisabled is true. So with that in mind, it makes sense to configure options to use callbacks to return focus based on whether it was closed via keyboard vs mouse. This would at least untangle the logic that is hard to follow, not knowing the context. 


https://github.com/Esri/calcite-design-system/pull/11625/files#diff-42c23e5a429472ae2ecb636c2ba383162681d40c5ff70c3f89be55d376052e32R137

**Comment by jcfranco** [comment]
@Elijbet Sorry for the delayed reply.  I believe https://github.com/Esri/calcite-design-system/pull/11823 addressed the outside click vs `Escape` closing issue, so this may no longer be valid. Could you take another look and confirm?
